### PR TITLE
feat(action): MZI-164 add task and action model

### PR DESCRIPTION
**Pull request by aymericdevriesere**
## Describe your changes
Adding tasks and action models
## Issue ticket number and link
[ MZI-164 ]
https://jira.support-ent.fr/browse/MZI-164
## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [ ] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: zimbra/src/main/java/fr/openent/zimbra/model/action/Action.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+import java.util.*;
+
+public class Action {
+    protected long id;
+    protected UUID userId;
+    protected Date createdAt;
+    protected ActionType actionType;
+    protected boolean approved;
+
+    Set<Task> tasks;
```
Comment (Matiyeu) [inline zimbra/src/main/java/fr/openent/zimbra/model/action/Action.java]:
```suggestion
    protected Set<Task> tasks;
```

File: zimbra/src/main/java/fr/openent/zimbra/model/action/Action.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        if (!JSONContainsActionData(actionData)) {
+            throw new Exception(String.format("[Zimbra@%s::Action] Json does not match Action model", Action.class));
+        }
+        long id = actionData.getInteger(Field.ACTION_ID);
+        try {
+            UUID userId = UUID.fromString(actionData.getString(Field.USER_ID));
+            Date created_at = Date.from(Instant.parse(actionData.getString(Field.CREATED_AT)));
+            ActionType actionType = ActionType.fromString(actionData.getString(Field.TYPE));
+            boolean approved = actionData.getBoolean(Field.APPROVED);
+            this.id = id;
```
Comment (Matiyeu) [inline zimbra/src/main/java/fr/openent/zimbra/model/action/Action.java]:
```suggestion
        try {
            UUID userId = UUID.fromString(actionData.getString(Field.USER_ID));
            Date created_at = Date.from(Instant.parse(actionData.getString(Field.CREATED_AT)));
            ActionType actionType = ActionType.fromString(actionData.getString(Field.TYPE));
            boolean approved = actionData.getBoolean(Field.APPROVED);
            this.id = actionData.getInteger(Field.ACTION_ID);
```

File: zimbra/src/main/java/fr/openent/zimbra/service/impl/QueueServiceImpl.java [hunk 3]
```diff
@@ -83,8 +89,8 @@ public void createActionInQueue(UserInfos user, String type, Boolean approved, H
         StringBuilder query = new StringBuilder();
 
         query.append("INSERT INTO ").append(actionTable)
-                .append(" (user_id, date, type, approved) ").append( "VALUES (?, ?, ?, ?)")
-                .append("RETURN id");
+                .append(" (" + Field.USER_ID + "," + Field.USER_ID + "," + Field.TYPE + "," + Field.APPROVED + ") ").append("VALUES (?, ?, ?, ?)")
```
Comment (Matiyeu) [inline zimbra/src/main/java/fr/openent/zimbra/service/impl/QueueServiceImpl.java:92]:
```suggestion
                .append(" (" + Field.USER_ID + ", " + Field.TYPE + ", " + Field.APPROVED + ") ").append("VALUES (?, ?, ?)")
```

Comment (aymericdevriesere) [reply zimbra/src/main/java/fr/openent/zimbra/service/impl/QueueServiceImpl.java:92]: Va être modif sur la 166 avec la modif des ces classes

**Review by Matiyeu — APPROVED** [review]

**Pull request closed without merging**